### PR TITLE
Shop (and other demo) touchups

### PR DESCRIPTION
**Pull request by chuck-dbos**
Cleanup installment 1.
* Update a lot of dependencies.
* Get rid of a few dependencies that we were not using.
* Set up lint / test in a few places.
* Add a github action to run shop (no tests yet, just the migrations)

More to follow, but this seems a simple increment and forward progress.

### CODE REVIEW

File: .github/workflows/shop.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      - 'bank/**'
+      - '.github/workflows/shop.yml'
+  pull_request:
+    types:
+      - ready_for_review
+      - opened
+      - reopened
+      - synchronize
+    paths:
+      - 'bank/**'
```
Comment (kraftp) [inline .github/workflows/shop.yml:16]: Shouldn't this be `e-commerce/**'?  Or am I misunderstanding it?

**Review by kraftp — APPROVED** [review]

File: e-commerce/payment-backend/package.json [hunk 2]
```diff
@@ -4,21 +4,25 @@
...
     "db:setup": "npx knex migrate:up",
-    "test": "npx knex migrate:rollback && npx knex migrate:up && jest"
+    "test": "npx knex migrate:rollback && npx knex migrate:up # && jest",
+    "lint": "eslint src",
+    "lint-fix": "eslint --fix src"
   },
   "devDependencies": {
-    "@dbos-inc/dbos-openapi": "^0.8.41-preview",
+    "@dbos-inc/dbos-cloud": "^0.8.61-preview",
+    "@dbos-inc/dbos-openapi": "^0.8.61-preview",
```
Comment (maxdml) [inline e-commerce/payment-backend/package.json:13]: Do we want `0.8.62` ?

File: .github/workflows/shop.yml (hunk 1 above)
Comment (chuck-dbos) [reply .github/workflows/shop.yml]: Yes,

File: e-commerce/payment-backend/package.json (hunk 2 above)
Comment (chuck-dbos) [reply e-commerce/payment-backend/package.json:13]: Right, I mentioned that at the meeting yesterday but now they are published and I will change it...

**Pull request merged**